File: backend/src/main/java/com/example/historyrag/feature/auth/AuthService.java
package com.example.historyrag.feature.auth;

import com.example.historyrag.feature.auth.dto.AuthUserResponse;
import com.example.historyrag.feature.auth.dto.LoginRequest;
import com.example.historyrag.feature.auth.dto.LoginResponse;
import com.example.historyrag.feature.auth.dto.RegisterRequest;
import com.example.historyrag.feature.auth.dto.RegisterResponse;

public interface AuthService {

    LoginResponse login(LoginRequest request, String deviceInfo, String ipAddress);

    RegisterResponse register(RegisterRequest request);

    LoginResponse refresh(String rawRefreshToken);

    void logout(String rawRefreshToken);

    AuthUserResponse getMe(String email, String accountType);
}

File: frontend/src/services/authService.js
import apiClient from './http/apiClient';
import { unwrap } from './http/response';
import { ENDPOINTS } from './endpoints';

const authService = {
    async login(credentials) {
        const res = await apiClient.post(ENDPOINTS.AUTH.LOGIN, credentials);
        return unwrap(res);
    },

    async register(payload) {
        const res = await apiClient.post(ENDPOINTS.AUTH.REGISTER, payload);
        return unwrap(res);
    },

    async logout() {
        const res = await apiClient.post(ENDPOINTS.AUTH.LOGOUT);
        return unwrap(res);
    },

    async getMe() {
        const res = await apiClient.get(ENDPOINTS.AUTH.ME);
        return unwrap(res);
    },
};

export default authService;